contracts/0.4.24/Lido.sol

```diff
@@ -79,33 +97,145 @@ contract Lido is ILido, IsContract, StETH, AragonApp {
     /// @dev number of Lido's validators available in the Beacon state
     bytes32 internal constant BEACON_VALIDATORS_POSITION = keccak256("lido.Lido.beaconValidators");
 
+    /// @dev percent in basis points of total pooled ether allowed to withdraw from MevTxFeeVault per LidoOracle report
+    bytes32 internal constant MEV_TX_FEE_WITHDRAWAL_LIMIT_POINTS = keccak256("lido.Lido.mevTxFeeWithdrawalLimitPoints");
+
+    /// @dev Just a counter of total amount of MEV and transaction rewards received by Lido contract
+    /// Not used in the logic
+    bytes32 internal constant TOTAL_MEV_TX_FEE_COLLECTED_POSITION = keccak256("lido.Lido.totalMevTxFeeCollected");
+
     /// @dev Credentials which allows the DAO to withdraw Ether on the 2.0 side
     bytes32 internal constant WITHDRAWAL_CREDENTIALS_POSITION = keccak256("lido.Lido.withdrawalCredentials");
 
     /**
     * @dev As AragonApp, Lido contract must be initialized with following variables:
-    * @param depositContract official ETH2 Deposit contract
+    * @param _depositContract official ETH2 Deposit contract
     * @param _oracle oracle contract
     * @param _operators instance of Node Operators Registry
+    * @param _treasury contract which accumulates treasury fee
+    * @param _insuranceFund contract which accumulates insurance fee
     */
     function initialize(
-        IDepositContract depositContract,
+        IDepositContract _depositContract,
         address _oracle,
         INodeOperatorsRegistry _operators,
         address _treasury,
         address _insuranceFund
     )
         public onlyInit
     {
-        _setDepositContract(depositContract);
-        _setOracle(_oracle);
-        _setOperators(_operators);
-        _setTreasury(_treasury);
-        _setInsuranceFund(_insuranceFund);
+        NODE_OPERATORS_REGISTRY_POSITION.setStorageAddress(address(_operators));
+        DEPOSIT_CONTRACT_POSITION.setStorageAddress(address(_depositContract));
+
+        _setProtocolContracts(_oracle, _treasury, _insuranceFund);
+
+        STAKE_LIMIT_POSITION.setStorageUint256(
```

Comment:
According to [lip-10](https://github.com/lidofinance/lido-improvement-proposals/blob/develop/LIPS/lip-10.md#initializer-vs-setter-functions) this kind of initialization should be done by introducing `initialize_vX()` and contract versioning

I also think that if we want to set the staking limit right after the upgrading, we also might want to set it explicitly on initialization after the deployment from scratch.

Reply:
Generally, I agree, but I propose to leave as is cause we have a contract's size issues :thinking: